io/shared/src/main/scala/org/scalajs/io/MemFiles.scala

```diff
@@ -10,52 +10,41 @@
 package org.scalajs.io
 
 import java.io._
-
 import java.nio.charset.StandardCharsets
 
-/** A base class for simple in-memory mutable virtual files. */
-class MemVirtualFile(val path: String) extends VirtualFile {
-  private[this] var _version: Option[String] = None
+/** A simple in-memory immutable virtual binary file. */
+class MemVirtualBinaryFile(
+    final val path: String,
+    final val content: Array[Byte],
+    final override val version: Option[String]
+) extends VirtualBinaryFile {
+  final def inputStream: InputStream = new ByteArrayInputStream(content)
+}
 
-  override def version: Option[String] = _version
-  def version_=(v: Option[String]): Unit = _version = v
+object MemVirtualBinaryFile {
+  def apply(path: String, content: Array[Byte],
+      version: Option[String] = None): MemVirtualBinaryFile = {
+    new MemVirtualBinaryFile(path, content, version)
+  }
 
-  final def withVersion(v: Option[String]): this.type = {
-    version = v
-    this
+  def fromStringUTF8(path: String, content: String,
+      version: Option[String] = None): MemVirtualBinaryFile = {
```

Comment:
Prefer overloads over default parameters. Default parameters cause trouble with binary compatibility.

Reply:
Also: `fromUTF8String` instead of `fromStringUTF8`?

Reply:
> fromUTF8String instead of fromStringUTF8

Not really if you think about it: the `String` is not UTF8, it's a `String`. The result is UTF8 :S

Reply:
Ah good point. My mistake.